src/Config/ProjectsConfiguration.php

```diff
@@ -52,6 +52,8 @@ public function getConfigTreeBuilder(): TreeBuilder
                                         ->arrayNode('tools')->prototype('scalar')->defaultValue([])->end()->end()
                                         ->arrayNode('php_extensions')->prototype('scalar')->defaultValue([])->end()->end()
                                         ->scalarNode('target_php')->defaultNull()->end()
+                                        ->scalarNode('custom_gitignore_part')->defaultNull()->end()
```

Comment:
Some option like panther, phpstan, psalm are on the global level, and some options like php are in the branches level. Some option like custom_gitignore_part are now on both.

It seems weird to have some github related workflow `phpstan` on the global level but not `frontend`.
I understand why you did this but we should have the same strategy for every option IMHO.

Should we 
1) Move the all options to the branches level ? It will be more verbose (with duplicate values) but more configurable.
2) Duplicate all the options on the branches level and the global level ? When it's the same value for every branches (for instance php version) it will avoid duplicates lines.  
3) Wait for the 4.0 release to introduce the frontend option (directly on the global level ?)

Also if I use `custom_gitignore_part` on the branch level I would have expect to override the project level.
So I would have not expect the code
```
{% if branch.customGitignorePart is not empty %}
{%- elseif project.customGitignorePart is not empty -%}
```
But only a check on `branch.customGitignorePart` with a getter returning `project.customGitignorePart` by default if the branch level option is not set.

Reply:
IMO we should introduce options as we need them, to avoid implementing things just for having them. Remember this code is not intended to be public used, just internal tool for sonata.

And about what happens if you define at branch level and at root level, I think it should take precedence the branch over the project configuration, I thought I have done that part right, the if will check first for branch config, and then for project config.

Reply:
Oh yes indeed, it an `elseif`.